### PR TITLE
job-manager / job-info: misc cleanup

### PR DESCRIPTION
**Pull request by chu11**
various cleanup / minor refactorings as I worked on #3233, #3209, #3234, #3245.

### CODE REVIEW

**Comment by garlick** [comment]
Suggestion (and apologies in advance if it seems like busy work):  could you rephrase commit messages using the `Problem:` format?   This will tend to leave a better history and force you to justify changes a bit more so IMHO is a good practice.  For example
```
job-manager: fix zlist_freefn call

The at_tail flag should be true after a zlist_append(), not false.
```
could be
```
job-manager: fix zlist_freefn call

Problem: zlist_free_fn() sets the destructor on the listeners list with at_tail=false,
which assumes listeners close to the beginning of the list are more likely to be
zlist_remove()d than those at the end of the list when the opposite is true.

Set at_tail=true here.
```
Except...seems like we probably don't expect enough listeners that this would matter, or know what order they will disconnect in.

**Comment by chu11** [comment]
> Except...seems like we probably don't expect enough listeners that this would matter, or know what order they will disconnect in.

I think the `at_tail` is only used for setting the freefn on a list item.  Its just so you don't have to scan the whole list for the item.

I've re-pushed doing the Problem / Solution style in a nice chunk of the commits.  I think some fixes are so trivial I left the commit messages as they were.

**Comment by garlick** [comment]
> I think the at_tail is only used for setting the freefn on a list item. Its just so you don't have to scan the whole list for the item.

Oh sorry, just double checked czmq source and you are correct.

**Review by garlick — APPROVED** [review]
Big improvement in the commit messages.  Thanks!
This LGTM!

**Comment by chu11** [comment]
hmmm one builder failed with

```
expecting success: 
	echo "[0-7]" >full3.exp &&
	flux start -s8 ${ARGS} ./qget.sh >full3.out &&
	test_cmp full3.exp full3.out

flux-start: 3 (pid 25752) Killed
flux-start: 1 (pid 25741) Killed
flux-start: 0 (pid 25740) Killed
not ok 4 - quorum reached on instance with 3 TBON levels
```

which also happened on #3238.  hard to imagine that any of my cleanup is involved.  i'll create an issue.  restarting that builder.

**Comment by chu11** [comment]
@garlick I assume MWP ok?

**Comment by garlick** [comment]
Go for it (MWP)!
